src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java

```diff
@@ -485,14 +485,13 @@ private void checkCheckClosingParens(final DetailAST ast) {
             final DetailAST paren = ast.getLastChild();
             final boolean parenExists = paren.getType() == TokenTypes.RPAREN;
 
-            if (closingParens == ClosingParens.ALWAYS
-                && !parenExists) {
-                log(ast.getLineNo(), MSG_KEY_ANNOTATION_PARENS_MISSING);
+            if (closingParens == ClosingParens.ALWAYS) {
+                if (!parenExists) {
+                    log(ast.getLineNo(), MSG_KEY_ANNOTATION_PARENS_MISSING);
+                }
             }
-            else if (closingParens == ClosingParens.NEVER
-                     && parenExists
+            else if (parenExists
                      && !ast.branchContains(TokenTypes.EXPR)
-                     && !ast.branchContains(TokenTypes.ANNOTATION_MEMBER_VALUE_PAIR)
```

Comment:
Branch contains is very powerful hence why we are trying to remove it.
We already know this is an annotation with a parenthesis, and branch contains scans the entire children and sub-children of the annotation so of course the annotation will have an expression if it has a member value pair hence why this is redundant.
Regression showed no differeces to try and keep this.